x-pack/legacy/plugins/security/server/lib/authorization/check_saved_objects_privileges.test.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SpacesPlugin } from '../../../../spaces';
+import { OptionalPlugin } from '../../../../../server/lib/optional_plugin';
+import { checkSavedObjectsPrivilegesWithRequestFactory } from './check_saved_objects_privileges';
+
+test(`checkPrivileges.atSpace when spaces is enabled`, async () => {
+  const expectedResult = Symbol();
+  const spaceId = 'foo-space';
+  const mockCheckPrivileges = {
+    atSpace: jest.fn().mockReturnValue(expectedResult),
+  };
+  const mockCheckPrivilegesWithRequest = jest.fn().mockReturnValue(mockCheckPrivileges);
+
+  const mockSpaces = {
+    isEnabled: true,
+    getSpaceId: jest.fn(),
+    spaceIdToNamespace: jest.fn(),
+    namespaceToSpaceId: jest.fn().mockReturnValue(spaceId),
+    getBasePath: jest.fn(),
+    getScopedSpacesClient: jest.fn(),
+  } as OptionalPlugin<SpacesPlugin>;
+  const request = Symbol();
+
+  const privilegeOrPrivileges = ['foo', 'bar'];
+
+  const checkSavedObjectsPrivileges = checkSavedObjectsPrivilegesWithRequestFactory(
+    mockCheckPrivilegesWithRequest,
+    mockSpaces
+  )(request as any);
+
+  const result = await checkSavedObjectsPrivileges(privilegeOrPrivileges);
+
```

Comment:
Can we add an `expect(mockSpaces.namespaceToSpaceId).toBeCalledWith(namespace)` after explicitly passing a namespace to `checkSavedObjectsPrivileges`?